### PR TITLE
add dist distillation loss

### PR DESCRIPTION
**Pull request by LKJacky**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

add dist distillation loss

### CODE REVIEW

File: mmrazor/models/losses/dist_loss.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    return inter_class_relation(y_s.transpose(0, 1), y_t.transpose(0, 1))
+
+
+@MODELS.register_module()
+class DISTLoss(nn.Module):
+
+    def __init__(
+        self,
+        beta=1.0,
+        gamma=1.0,
```
Comment (HIT-cwh) [inline mmrazor/models/losses/dist_loss.py:31]: Maybe it's more explicit to use `inter_loss_weight` and `intra_loss_weight` instead, and delete `loss_weight` attribute.

File: mmrazor/models/losses/dist_loss.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        self.gamma = gamma
+        self.tau = tau
+
+        self.loss_weight = loss_weight
+        self.teacher_detach = teacher_detach
+
+    def forward(self, preds_S, preds_T: torch.Tensor):
+        if self.teacher_detach:
+            preds_T = preds_T.detach()
+        y_s = (preds_S / self.tau).softmax(dim=1)
```
Comment (HIT-cwh) [inline mmrazor/models/losses/dist_loss.py:47]: Maybe the name `logits_S` is better.

File: mmrazor/models/losses/dist_loss.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+from mmrazor.registry import MODELS
+
+
+def cosine_similarity(a, b, eps=1e-8):
+    return (a * b).sum(1) / (a.norm(dim=1) * b.norm(dim=1) + eps)
+
+
+def pearson_correlation(a, b, eps=1e-8):
+    return cosine_similarity(a - a.mean(1).unsqueeze(1),
```
Comment (HIT-cwh) [inline mmrazor/models/losses/dist_loss.py:13]: Maybe it's more normal to replace `a.mean(1).unsqueeze(1)` with `a.mean(dim=1, keepdim=True)`

File: .gitignore [hunk 4]
```diff
@@ -11,7 +11,7 @@ __pycache__/
 .Python
 build/
 develop-eggs/
-dist/
+/dist
```
Comment (HIT-cwh) [inline .gitignore:14]: Maybe this line can be deleted.

File: configs/distill/mmcls/dist/dist_logits_resnet34_resnet18_8xb32_in1k.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    'mmcls::_base_/default_runtime.py'
+]
+
+teacher_ckpt = 'https://download.openmmlab.com/mmclassification/v0/resnet/resnet34_8xb32_in1k_20210831-f257d4e6.pth'  # noqa: E501
+
+model = dict(
+    _scope_='mmrazor',
+    type='SingleTeacherDistill',
+    data_preprocessor=dict(
+        type='ImgDataPreprocessor',
```
Comment (HIT-cwh) [inline configs/distill/mmcls/dist/dist_logits_resnet34_resnet18_8xb32_in1k.py:13]: I'm not sure whether we have to use `mmcls.ImgDataPreprocessor`.

Comment (LKJacky) [reply configs/distill/mmcls/dist/dist_logits_resnet34_resnet18_8xb32_in1k.py:13]: YES，it is used to prepare input images

Comment (HIT-cwh) [reply configs/distill/mmcls/dist/dist_logits_resnet34_resnet18_8xb32_in1k.py:13]: I mean whether we have to use `mmcls.ImgDataPreprocessor` but not `ImgDataPreprocessor`. As the default scope is mmrazor, if we want to use ImgDataPreprocessor in mmcls, maybe we have to add `mmcls.` prefix. But I'm not sure.

Comment (LKJacky) [reply configs/distill/mmcls/dist/dist_logits_resnet34_resnet18_8xb32_in1k.py:13]: "mmcls." is not necessary, the config is  excutable.

**Review by pppppM — APPROVED** [review]
